src/frontend/planner_test/tests/testdata/output/union.yaml

```diff
@@ -292,3 +309,129 @@
     └─BatchHashAgg { group_key: [1:Int32], aggs: [] }
       └─BatchExchange { order: [], dist: HashShard(1:Int32) }
         └─BatchValues { rows: [[1:Int32], [2:Int32], [3:Int32], [4:Int32], [5:Int32], [5:Int32]] }
+- name: test merged union stream key
+  sql: |
+    create table t1 (a int, b numeric, c bigint);
+    create table t2 (a int, b numeric, c bigint);
+    create table t3 (a int, b numeric, c bigint);
+    create table t4 (a int, b numeric, c bigint);
+    create table t5 (a int, b numeric, c bigint);
+    select * from t1 union all select * from t2 union all select * from t3 union all select * from t4 union all select * from t5;
+  batch_plan: |-
+    BatchUnion { all: true }
+    ├─BatchExchange { order: [], dist: Single }
+    │ └─BatchScan { table: t1, columns: [t1.a, t1.b, t1.c], distribution: SomeShard }
+    ├─BatchExchange { order: [], dist: Single }
+    │ └─BatchScan { table: t2, columns: [t2.a, t2.b, t2.c], distribution: SomeShard }
+    ├─BatchExchange { order: [], dist: Single }
+    │ └─BatchScan { table: t3, columns: [t3.a, t3.b, t3.c], distribution: SomeShard }
+    ├─BatchExchange { order: [], dist: Single }
+    │ └─BatchScan { table: t4, columns: [t4.a, t4.b, t4.c], distribution: SomeShard }
+    └─BatchExchange { order: [], dist: Single }
+      └─BatchScan { table: t5, columns: [t5.a, t5.b, t5.c], distribution: SomeShard }
+  stream_plan: |-
+    StreamMaterialize { columns: [a, b, c, t1._row_id(hidden), 0:Int32(hidden)], stream_key: [t1._row_id, 0:Int32], pk_columns: [t1._row_id, 0:Int32], pk_conflict: NoCheck }
```

Comment:
Please forgive my nagging... Is it possible to give the `0:Int32` column a meaningful name such as `$src` etc.